feat: implement horizontal footer layout with modern social media icons

Recxsmacx opened the pull request:
Related to https://github.com/asyncapi/conference-website/issues/659

Enhance the social section of the website to improve user engagement and interactivity.

Footer Component Changes:

Replace the hardcoded social media links with a new Socials component.
Add hover effects and tooltips for social media icons.
Add onClick event handlers to redirect users to respective social media pages.
Increase the gap between the "Code of Conduct" link and the line in the footer by adding margin to the a element.
Socials Component:

Create a new Socials component to handle social media icons.
Add LinkedIn, GitHub, Twitter, and YouTube icons with respective onClick event handlers.
Add tooltips for each social media icon to provide additional context.
Add hover effects to make the icons more visually appealing.

Recxsmacx committed:
feat: implement horizontal footer layout with modern social media icons

- Refactor footer to use consistent horizontal layout across all screen sizes
- Create new Socials component with custom styled circular icons
- Add interactive tooltips with platform-specific hover colors
- Update Cypress tests to include social media icon testing
- Add CSS styles for tooltip animations and hover effects
- Replace image-based icons with inline SVG for better performance

Changes:
- components/Footer/footer.tsx: horizontal layout with flex-wrap
- components/Footer/socials.tsx: new component with LinkedIn, GitHub, Twitter, YouTube
- cypress/e2e/Footer.cy.ts: enhanced test suite with social media tests
- styles/globals.css: tooltip styles and hover animations

Recxsmacx committed:
fix: add CSS support for tooltip visibility in Cypress tests

- Add .show-for-test class support to make tooltips visible during testing
- Use !important to override default opacity: 0 for test scenarios
- All 12 Cypress tests now passing including tooltip visibility tests

AceTheCreator reviewed (changes requested):
Looks really, @Recxsmacx. Can you make it responsive on small screen?

Recxsmacx committed:
feat: make footer responsive for mobile devices

Recxsmacx committed:
feat: make footer responsive for mobile devices

AceTheCreator reviewed (changes requested):
@Recxsmacx, what I meant by making it responsive is simply adjusting the footer layout to collapse from a row into a column on smaller screens, not just making the text smaller.

Recxsmacx committed:
Update footer.tsx with social components

Recxsmacx commented:
> @Recxsmacx, what I meant by making it responsive is simply adjusting the footer layout to collapse from a row into a column on smaller screens, not just making the text smaller.

@AceTheCreator I kept it the old way on smaller screens the code of conduct button vanishes to keep the footer look clean and not messy wdyt?
rest the responsiveness is done

AceTheCreator reviewed (changes requested):
Remove unnecessary code formatter changes in this file

AceTheCreator commented:
Kindly resolve the conflicts, so we can merge it

AceTheCreator reviewed (changes requested):
discard changes made under  cypress/download

Recxsmacx committed:
Merge latest master changes

- Resolved conflicts in footer.tsx by keeping Socials component while using master's import style
- Resolved conflicts in Footer.cy.ts by integrating close-button clicks from master with social media tooltip tests
- All functionality preserved from both branches

Recxsmacx committed:
fix: Add JSX import to socials component and install react-dom types

- Fix TypeScript error in socials.tsx by importing JSX
- Install @types/react-dom to resolve popup component build error
- Maintain functionality while ensuring type safety

Recxsmacx committed:
feat: Add AsyncAPI Conference 2025 PDF document

- Add conf 2025.pdf to cypress/Downloads directory
- Document contains conference information and schedule
- Referenced in Cypress tests for download functionality

Recxsmacx committed:
Merge branch 'master' into Footer-socials

Recxsmacx committed:
chore: sync package-lock.json with package.json

Recxsmacx committed:
Merge branch 'master' into Footer-socials

AceTheCreator committed:
Merge branch 'master' into Footer-socials

AceTheCreator committed:
Merge branch 'master' into Footer-socials

AceTheCreator commented on package.json:
unnecessary code changes here. kindly remove

AceTheCreator reviewed (changes requested):
Left you some reviews @Recxsmacx

Recxsmacx committed:
Merge branch 'master' into Footer-socials

Recxsmacx committed:
applied for each in cypress tests

Recxsmacx committed:
Merge branch 'Footer-socials' of https://github.com/Recxsmacx/conference-website into Footer-socials

Recxsmacx committed:
Delete cypress/Downloads/conf 2025.pdf

Recxsmacx committed:
imported pre available socials svgs

Recxsmacx committed:
Merge branch 'Footer-socials' of https://github.com/Recxsmacx/conference-website into Footer-socials

Recxsmacx committed:
dependency issue resolved

Recxsmacx committed:
Merge branch 'master' into Footer-socials

Recxsmacx committed:
Merge branch 'master' into Footer-socials

thulieblack committed:
Merge branch 'master' into Footer-socials

AceTheCreator commented on styles/globals.css:
```diff
@@ -14,7 +14,65 @@ body {
+  background: #ff0000;
 }
 
 .text-gradient {
```
I noticed a few CSS changes in your PR that probably shouldn’t be there. It appears that this may have been introduced by an auto-formatter in your IDE.

princerajpoot20 replied:
@Recxsmacx We should not add any unnecessary lines. This makes the code difficult to maintain in the long run. Please go ahead and remove that line and push your changes.

AceTheCreator reviewed (changes requested):
Left you some final review @Recxsmacx

princerajpoot20 commented on styles/globals.css:
```diff
@@ -61,198 +119,165 @@ body {
+.card-opacity:hover{
   opacity: 100;
 }
 
```
Please remove all the unnecessary formatting changes.
The existing formatting is good.

We should not commit these formatting changes just because your local formatter suggests them. Someone else may have different settings and might change them again, which makes code reviews unnecessarily difficult. Hence, please stick to the original formatting unless you have a strong reason to change it.

Thanks.

Recxsmacx replied:
@princerajpoot20 solved most of the formatting issues in global css file please have a look on it

princerajpoot20 reviewed (changes requested):
Please have a look at the comment.

Recxsmacx committed:
Merge branch 'master' into Footer-socials

Recxsmacx committed:
Merge branch 'master' into Footer-socials

Recxsmacx committed:
fixed formatting issues

Recxsmacx committed:
Merge branch 'master' into Footer-socials

Recxsmacx commented:
@princerajpoot20 can you check this!! resolved the possible formatting changes cc:- @AceTheCreator

Recxsmacx committed:
Merge branch 'master' into Footer-socials

thulieblack committed:
Merge branch 'master' into Footer-socials

Recxsmacx committed:
Merge branch 'master' into Footer-socials

Recxsmacx committed:
Merge branch 'master' into Footer-socials

Recxsmacx committed:
Merge branch 'master' into Footer-socials

AceTheCreator reviewed (changes requested):
Left you some review @Recxsmacx

Recxsmacx committed:
Merge branch 'master' into Footer-socials

Recxsmacx committed:
applie data-test across the footer file and the cy file

Recxsmacx committed:
Merge branch 'Footer-socials' of https://github.com/Recxsmacx/conference-website into Footer-socials

Recxsmacx committed:
cypress fixed

Recxsmacx committed:
Merge branch 'master' into Footer-socials

AceTheCreator committed:
Merge branch 'master' into Footer-socials

AceTheCreator reviewed (approved):
LGTM! 🚀
